packages/language-service/src/completions.ts

```diff
@@ -528,59 +551,106 @@ class ExpressionVisitor extends NullTemplateVisitor {
    *
    * @param attr descriptor for attribute name and value pair
    * @param binding template binding for the expression in the attribute
+   * @param directiveMetadata CompileDirectiveSummary
    */
-  private microSyntaxInAttributeValue(attr: AttrAst, binding: TemplateBinding) {
-    const key = attr.name.substring(1);  // remove leading asterisk
-
-    // Find the selector - eg ngFor, ngIf, etc
-    const selectorInfo = getSelectors(this.info);
-    const selector = selectorInfo.selectors.find(s => {
-      // attributes are listed in (attribute, value) pairs
-      for (let i = 0; i < s.attrs.length; i += 2) {
-        if (s.attrs[i] === key) {
-          return true;
-        }
-      }
-    });
-
-    if (!selector) {
-      return;
-    }
-
+  private microSyntaxInAttributeValue(
+      attr: AttrAst, binding: TemplateBinding, directiveMetadata: CompileDirectiveSummary) {
     const valueRelativePosition = this.position - attr.sourceSpan.start.offset;
 
+    const bindingSource = attr.value.substring(binding.span.start, binding.span.end);
+    const endOfSyntax = valueRelativePosition > binding.span.start + bindingSource.trim().length;
+
     if (binding.keyIsVar) {
-      const equalLocation = attr.value.indexOf('=');
-      if (equalLocation > 0 && valueRelativePosition > equalLocation) {
-        // We are after the '=' in a let clause. The valid values here are the members of the
-        // template reference's type parameter.
-        const directiveMetadata = selectorInfo.map.get(selector);
-        if (directiveMetadata) {
-          const contextTable =
-              this.info.template.query.getTemplateContext(directiveMetadata.type.reference);
-          if (contextTable) {
-            // This adds symbols like $implicit, index, count, etc.
-            this.addSymbolsToCompletions(contextTable.values());
-            return;
-          }
+      const equalLocation = bindingSource.indexOf('=');
+      const asLocation = bindingSource.indexOf(' as ');
+      if (equalLocation > 0 && valueRelativePosition > equalLocation + binding.span.start ||
+          (asLocation > 0 && asLocation + binding.span.start >= valueRelativePosition)) {
+        // We are after the '=' or before the 'as' in a let clause. The valid values here are the
+        // members of the template reference's type parameter.
```

Comment:
```suggestion
        // members of the directive's template context.
```

would you mind changing this while you're here? i think the original comment obscures what we're trying to do.